src/Adapters/EventSource.Netstandard13/EventSourceTelemetryModule.cs

```diff
@@ -130,7 +137,11 @@ protected override void OnEventWritten(EventWrittenEventArgs eventData)
             {
                 try
                 {
-                    this.onEventWrittenHandler(eventData, this.client);
+                    // We can't deal with disabled sources until event is raised due to: https://github.com/dotnet/coreclr/issues/14434
+                    if (DisabledSources.Count == 0 || !IsDroppingEvent(eventData))
```

Comment:
why not move the first condition inside the method?

Reply:
Thanks for the ask. Here's my consideration:

The assumptions:
1. This 'OnEventWritten' will be invoked huge amount of times per second.
2. DisabledSources's largely going to be empty.

Based on those, I want to reduce any context switch for the execution of the method whenever possible. Putting a pre-check, hopefully, it returns true, to avoid switching call-context to 'IsDroppingEvent' is the main reason.

---
Now that I think about it, having a pre-calculated list count shall further improve the performance and I'll update the code accordingly.